packages/donations-app/devServer/controller/donation.js

```diff
@@ -123,9 +126,42 @@ module.exports = function (app) {
     return res.json(donations);
   }));
 
+  async function redirectRequestToAllExpEnvironments(req) {
+    const environments = await Environment.find({});
+    for (const environment of environments) {
+      const urlParsed = url.parse(req.url);
+      const newUrl = environment.url + urlParsed.pathname + urlParsed.search;
+      console.log('hookRedirection: Resending to ' + newUrl);
+      await new Promise((resolve) => {
+        let req = http.get(newUrl, (res) => {
+          console.log(`hookRedirection: Got reponse from ${environment.url}, statusCode: ${res.statusCode}`);
+          resolve();
+        });
+
+        req.on('timeout', () => {
+          console.log('hookRedirection: Time is out ' + environment.url);
+          req.abort();
+          resolve();
```

Comment:
It won't change anything in this context so I don't insist, but let's reject promises when it's not successful for consistent error handling

Reply:
Ok, I see your point with mangoPay hook errors.